### PR TITLE
Enable Istio Agent DNS interception for preview

### PR DESCRIPTION
**Pull request by rshriram**
Per discussion in Networking WG.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by linsun** [comment]
This LGTM.  I think @howardjohn had some concerns with enabling for demo profile.  If we have tests for demo that covers bookinfo, I think it 'd be good.

Bonus - do we have any docs on how to use this?  Would be good to clarify that with this change, the following config (within SE?) isn't necessary.

**Comment by rshriram** [comment]
>This LGTM. I think @howardjohn had some concerns with enabling for demo profile. If we have tests for demo that covers bookinfo, I think it 'd be good.

DNS is enabled for all tests in Istio (including postsubmit). So I guess bookinfo is covered?

With regard to Docs, yes its on my plate to describe why certain SE hacks like distinct tcp ports are unnecessary.

File: releasenotes/notes/agent-dns-capture.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: release-notes/v2
+kind: feature
+area: networking
+issue:
+- 26826
+releaseNotes:
+- |
+  *Added* Experimental DNS resolution in the Istio agent to better support use cases like service entries
```
Comment (brian-avery) [inline releasenotes/notes/agent-dns-capture.yaml:8]:
```suggestion
  **Added** Experimental DNS resolution in the Istio agent to better support use cases like service entries
```

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ var (
 	skipParseTokenEnv = env.RegisterBoolVar("SKIP_PARSE_TOKEN", false,
 		"Skip Parse token to inspect information like expiration time in proxy. This may be possible "+
 			"for example in vm we don't use token to rotate cert.").Get()
-	proxyXDSViaAgent = env.RegisterStringVar("ISTIO_META_PROXY_XDS_VIA_AGENT", "",
+	proxyXDSViaAgent = env.RegisterStringVar("ISTIO_META_PROXY_XDS_VIA_AGENT", "true",
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go:153]: Should be more cautious about this, it can be severe if there is any issue.

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/main.go:153]: Can we add `ISTIO_META_PROXY_XDS_VIA_AGENT` to those two profiles for now? I think we still need

**Review by howardjohn — COMMENTED** [review]
Adding to preview profile: LGTM, preview profile is for users who want bleeding edge and are ok with things breaking or being a bit "rough". They are explicitly opting into this

Adding to demo profile: -1 for me. Demo users are people brand new to Istio. If they have rough edges or bugs they will be hopeless and likely ditch Istio. Demo should be MOST stable profile, not our least, IMO.

**Comment by rshriram** [comment]
done

File: manifests/profiles/preview.yaml [hunk 3]
```diff
@@ -4,6 +4,15 @@
 apiVersion: install.istio.io/v1alpha1
 kind: IstioOperator
 spec:
+  meshConfig:
+    accessLogFile: /dev/stdout
```
Comment (howardjohn) [inline manifests/profiles/preview.yaml:8]: Not sure this should be set here?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**